### PR TITLE
Backport OpenMPI variables from `khiops-env` to mitigate errors

### LINKED ISSUE #196: Backport OpenMPI Fixes for Rocky and Docker from Khiops
#### Description
PR https://github.com/KhiopsML/khiops/pull/313 added two environment variables to the `khiops-env` scripts for OpenMPI, in order to mitigate issue https://github.com/KhiopsML/khiops/issues/307, but also a different warning on Docker deployments.

These two variables need to be added in the `KhiopsLocalRunner` for "binary+pip" installs:

- `OMPI_MCA_btl_vader_single_copy_mechanism=none` in all cases
- `PSM3_DEVICES=self` for Rocky Linux only

### PR DESCRIPTION
**Pull request by popescu-v**
closes #196

### CODE REVIEW

File: khiops/core/internals/runner.py [hunk 1]
```diff
@@ -294,6 +294,63 @@ def _check_executable(bin_path):
...
+    with open(os_release_file_path, encoding="ascii") as os_release_info:
+        for entry in os_release_info:
+            if entry.startswith("NAME"):
+                linux_distribution = entry.split("=")[-1].strip('"\n')
+                break
+
+    # The `NAME` variable is always defined according to the freedesktop.org
+    # standard:
+    # https://www.freedesktop.org/software/systemd/man/latest/os-release.html
+    return linux_distribution
```
Comment (folmos-at-orange) [inline khiops/core/internals/runner.py:307]: What happens if the file doesn't exist?

Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:307]: Maybe add `assert platform.system() == "Linux"` ?

File: khiops/core/internals/runner.py [hunk 2]
```diff
@@ -294,6 +294,63 @@ def _check_executable(bin_path):
...
+def _get_linux_distribution_from_os_release_file(os_release_file_path):
+    with open(os_release_file_path, encoding="ascii") as os_release_info:
+        for entry in os_release_info:
+            if entry.startswith("NAME"):
+                linux_distribution = entry.split("=")[-1].strip('"\n')
+                break
+
+    # The `NAME` variable is always defined according to the freedesktop.org
+    # standard:
+    # https://www.freedesktop.org/software/systemd/man/latest/os-release.html
```
Comment (folmos-at-orange) [inline khiops/core/internals/runner.py:306]: Move this above the code that does the work.

File: khiops/core/internals/runner.py [hunk 3]
```diff
@@ -294,6 +294,63 @@ def _check_executable(bin_path):
         )
 
 
+def _get_linux_distribution_from_os_release_file(os_release_file_path):
+    with open(os_release_file_path, encoding="ascii") as os_release_info:
```
Comment (folmos-at-orange) [inline khiops/core/internals/runner.py:298]: Rename to `os_release_info` to `os_release_info_file` to be coherent with the rest of codebase.

File: khiops/core/internals/runner.py [hunk 4]
```diff
@@ -294,6 +294,63 @@ def _check_executable(bin_path):
...
+                break
+
+    # The `NAME` variable is always defined according to the freedesktop.org
+    # standard:
+    # https://www.freedesktop.org/software/systemd/man/latest/os-release.html
+    return linux_distribution
+
+
+def detect_linux_distribution_name():
+    """Detect Linux distribution name
```
Comment (folmos-at-orange) [inline khiops/core/internals/runner.py:311]: I'd call this `get_linux_distribution_name`.

File: khiops/core/internals/runner.py [hunk 5]
```diff
@@ -294,6 +294,63 @@ def _check_executable(bin_path):
...
+    str
+        Name of the Linux distribution, converted to lowecase
+
+    Raises
+    ------
+    OSError
+        If neither `/etc/os-release` nor `/usr/lib/os-release` are found
+    """
+
+    python_ver_major, python_ver_minor, _ = platform.python_version_tuple()
```
Comment (folmos-at-orange) [inline khiops/core/internals/runner.py:327]: ~~Add comment to this.~~ merge this line to the paragraph below.

File: khiops/core/internals/runner.py [hunk 6]
```diff
@@ -294,6 +294,63 @@ def _check_executable(bin_path):
...
+    -------
+    str
+        Name of the Linux distribution, converted to lowecase
+
+    Raises
+    ------
+    OSError
+        If neither `/etc/os-release` nor `/usr/lib/os-release` are found
+    """
+
```
Comment (folmos-at-orange) [inline khiops/core/internals/runner.py:326]: I'd add `get_linux_distriubtion_from..` as a function defined within this one because it is used only here.

**Review by folmos-at-orange — CHANGES_REQUESTED** [review]

File: khiops/core/internals/runner.py (hunk 1 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:307]: If the file does not exist, then `FileNotFoundError` is raised. This exception is handled in the `detect_linux_distribution` function that calls this (module-"private") helper function.

File: khiops/core/internals/runner.py (hunk 2 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:306]: Yes, OK.

File: khiops/core/internals/runner.py (hunk 3 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:298]: Ok for `s/os_release_info/os_release_info_file/`.

File: khiops/core/internals/runner.py (hunk 4 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:311]: Yes, ok.

File: khiops/core/internals/runner.py (hunk 5 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:327]: Ok.

File: khiops/core/internals/runner.py (hunk 6 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:326]: Yes, ok, I had thought about it too.

File: khiops/core/internals/runner.py (hunk 1 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:307]: > Maybe add `assert platform.system() == "Linux"` ?

Ok, I'll add the `assert` to the (renamed) `get_linux_distribution` method, as this "private" method would become an inner method of the former.

Comment (popescu-v) [reply khiops/core/internals/runner.py]: Done.

File: khiops/core/internals/runner.py (hunk 2 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py]: Done.

File: khiops/core/internals/runner.py (hunk 3 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py]: Done.

File: khiops/core/internals/runner.py (hunk 4 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:298]: Done.

File: khiops/core/internals/runner.py (hunk 5 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py]: Done.

**Review by folmos-at-orange — APPROVED** [review]

**Pull request merged**